graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/IndicesAdapterOS2.java

```diff
@@ -431,20 +438,30 @@ public List<ShardsInfo> getShardsInfo(String indexName) {
         return catApi.getShardsInfo(indexName);
     }
 
+
     @Override
     public IndicesBlockStatus getIndicesBlocksStatus(final List<String> indices) {
         if (indices == null || indices.isEmpty()) {
             throw new IllegalArgumentException("Expecting list of indices with at least one index present.");
         }
-        final GetSettingsRequest getSettingsRequest = new GetSettingsRequest()
-                .indices(indices.toArray(new String[]{}))
-                .indicesOptions(IndicesOptions.fromOptions(false, true, true, true))
-                .names(new String[]{});
 
-        return client.execute((c, requestOptions) -> {
-            final GetSettingsResponse settingsResponse = c.indices().getSettings(getSettingsRequest, requestOptions);
-            return BlockSettingsParser.parseBlockSettings(settingsResponse);
-        });
+        if(String.join(",", indices).length() > MAX_INDICES_URL_LENGTH) {
+            final GetSettingsRequest getSettingsRequest = new GetSettingsRequest()
+                    .indicesOptions(IndicesOptions.fromOptions(false, true, true, true))
```

Comment:
Can we use `.names("index.blocks.read", ...)` to get only the blocks here?

Reply:
sure, but I'm going to change it in both places so that that part works the same in both cases